Mathlib/Tactic/FieldSimp.lean

```diff
@@ -0,0 +1,156 @@
+/-
+Copyright (c) 2019 Sébastien Gouëzel. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Sébastien Gouëzel, David Renshaw
+-/
+
+import Lean.Elab.Tactic.Basic
+import Lean.Meta.Tactic.Simp.Main
+import Std.Lean.Parser
+import Mathlib.Algebra.Group.Units
+
+/-!
+# `field_simp` tactic
+
+Tactic to clear denominators in algebraic expressions, based on `simp` with a specific simpset.
+-/
+
+namespace Mathlib.Tactic.FieldSimp
+
+open Lean Elab.Tactic Parser.Tactic Lean.Meta
+
+initialize registerTraceClass `Tactic.field_simp
+
+/-- Constructs a trace message for the `discharge` function. -/
+private def dischargerTraceMessage (prop: Expr) : Except ε (Option Expr) → SimpM MessageData
+| .error _ | .ok none => return m!"{crossEmoji} discharge {prop}"
+| .ok (some _) => return m!"{checkEmoji} discharge {prop}"
+
+/-- Discharge strategy for the field_simp tactic. -/
+partial def discharge (prop : Expr) : SimpM (Option Expr) :=
```

Comment:
I used the default discharger implementation as a model for this function. https://github.com/leanprover/lean4/blob/1f41b912064e49979bfefa5874ea53cb25934a06/src/Lean/Meta/Tactic/Simp/Main.lean#L815-L835